### PR TITLE
Add tox and django test framework

### PR DESCRIPTION
**Pull request by xuhang57**
Follow what horizon does upstream. Use tox and django test framework together for unit testing in our code-base.

WIP because I need to figure out how to set up `tox.ini` file properly. Thanks.

**Update**:
1. Have `tox.ini` file ready and tested it in my testing server.
2. Add setup.py as a config file for running `tox` command under `/opt/dev/atmosphere`

**Wishlist**

- [ ] Documentation about how to run `tox` and how to write unit test for GIJI-backend

**Task(s)**
- [ ] Reviewed by at least two engineers in the MOC team and one of them is Rob.

### CODE REVIEW

**Review by kamfonik — APPROVED** [review]
This looks good to me.  I haven't set up tox before, but I looked through the docs and what you have here makes sense.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  bash
+
+[testenv:py27dj19]
+commands =
+  pip install -U django>=1.9,<1.10
+  pip install factory_boy
+  pip install future
+  pip install vcrpy
+  pip install mock
+  {[unit_tests]commands}
```
Comment (rob-baron) [inline tox.ini]: Is there a reason why you did all of the pip installs as opposed to adding:
deps=djanog>=1.9,<1.10, factory_boy ...

**Review by rob-baron — CHANGES_REQUESTED** [review]
Is there a reason why you did the pip install [package] as opposed to adding a deps variable with the packages listed?  If there isn't one, this should be changed to a more standard way.

**Comment by xuhang57** [comment]
Found them in the `dev_requirements.txt`

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+       -r{toxinidir}/dev_requirements.txt
+commands =
+  find /opt/dev/atmosphere -type f -name "*.pyc" -delete
+whitelist_externals=
+  find
+  bash
+
+[testenv:py27dj19]
+commands =
+  pip install -U django>=1.9,<1.10
```
Comment (rob-baron) [inline tox.ini:20]: isn't this specified in the requirements.txt file on line 32?

**Review by rob-baron — CHANGES_REQUESTED** [review]
Check to see if django isn't already being included via the requirements.txt file - that way we can get rid of the command in the ini file.

File: tox.ini (hunk 2 above)
Comment (lucashxu) [reply tox.ini:20]: This is the standard way to do it. Potentially we need to test with different version of Django as the project grows.

**Review by rob-baron — APPROVED** [review]

**Pull request merged**